### PR TITLE
Support humble

### PR DESCRIPTION
**Pull request by tonynajjar**
## Fixes #
https://github.com/microsoft/Azure_Kinect_ROS_Driver/issues/256

### Description of the changes:
- Added defaults to declare_parameter as is required in Humble
- Added [angles](https://github.com/ros/angles) to Cmakelist to fix `fatal error: angles/angles.h: No such file or directory`
- Minor compiler warning fixes
- Removed "required" parameter as I don't see it used anywhere

### Required before submitting a Pull Request:
- [x] I reviewed [CONTRIBUTING.md](https://github.com/microsoft/Azure_Kinect_ROS_Driver/blob/master/CONTRIBUTING.md)
- [x] I [built my changes](https://github.com/microsoft/Azure_Kinect_ROS_Driver/blob/master/docs/building.md) locally
- [Not yet] I tested my changes with a device
- [Not Needed] I changed both the ROS1 and ROS2 branches 

### I tested changes on: 
- [Not planning to] Windows
- [x] Linux
- [ Not needed] ROS1
- [x] ROS2

### CODE REVIEW

File: src/k4a_ros_device.cpp [hunk 1]
```diff
@@ -58,30 +58,29 @@ K4AROSDevice::K4AROSDevice()
...
+  this->declare_parameter("body_tracking_smoothing_factor", rclcpp::ParameterValue(0.0f));
+  this->declare_parameter("rescale_ir_to_mono8", rclcpp::ParameterValue(false));
+  this->declare_parameter("ir_mono8_scaling_factor", rclcpp::ParameterValue(1.0f));
+  this->declare_parameter("imu_rate_target", rclcpp::ParameterValue(0));
+  this->declare_parameter("wired_sync_mode", rclcpp::ParameterValue(0));
+  this->declare_parameter("subordinate_delay_off_master_usec", rclcpp::ParameterValue(0));
+  // this->declare_parameter({depth_raw_topic + compressed_format});
+  // this->declare_parameter({depth_raw_topic + compressed_png_level});
+  // this->declare_parameter({depth_rect_topic + compressed_format});
+  // this->declare_parameter({depth_rect_topic + compressed_png_level});
```
Comment (tonynajjar) [inline src/k4a_ros_device.cpp]: I'm not really sure here, these were parameters with their name being the topic name itself? What is the use here? Where are these used?

Comment (ooeygui) [reply src/k4a_ros_device.cpp]: Honestly, not sure either. I think this was done as an alternative to topic remapping, but likely should be removed.

**Review by ooeygui — APPROVED** [review]
I'm approving this change. Thank you for the contribution!

**Comment by tonynajjar** [comment]
> Honestly, not sure either. I think this was done as an alternative to topic remapping, but likely should be removed.

thought so too, I'll remove then

**Review by ooeygui — APPROVED** [review]

**Review by ooeygui — APPROVED** [review]

**Comment by ooeygui** [comment]
Don't worry about CI - I'll have to switch it to humble when the Humble Windows release is completed.

**Comment by tonynajjar** [comment]
@microsoft-github-policy-service agree [company="{Pixel Robotics}"]

**Comment by tonynajjar** [comment]
> @tonynajjar please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> 
> Contributor License Agreement



@microsoft-github-policy-service agree company="Pixel Robotics"

**Comment by st99315** [comment]
Hi, Do you get the error following when execute the launch?
```bash
$ ros2 launch azure_kinect_ros_driver driver.launch.py

[INFO] [launch]: All log files can be found below /root/.ros/log/2022-11-03-07-52-45-234984-jun-15580
[INFO] [launch]: Default logging verbosity is set to INFO
Robot description xacro_file : /root/ws_moveit/install/azure_kinect_ros_driver/share/azure_kinect_ros_driver/urdf/azure_kinect.urdf.xacro
Robot description urdf_path : /root/ws_moveit/install/azure_kinect_ros_driver/share/azure_kinect_ros_driver/urdf/azure_kinect.urdf
[ERROR] [launch]: Caught exception in launch (see debug for traceback): package 'azure_kinect_ros_driver' found at '/root/ws_moveit/install/azure_kinect_ros_driver', but libexec directory '/root/ws_moveit/install/azure_kinect_ros_driver/lib/azure_kinect_ros_driver' does not exist
```

Finally, it can be executed after revised CMakeLists.txt 
```cmake
install(TARGETS ${PROJECT_NAME}_node
  RUNTIME DESTINATION lib/${PROJECT_NAME}/
)
```

**Comment by tonynajjar** [comment]
Thanks for the feedback, I haven't tried launching it yet actually as I didn't have a camera. I'll try it out and update you, at the latest next week.

**Comment by tonynajjar** [comment]
@st99315 I can't seem to reproduce your bug. Did you make sure to source setup.bash after building?

**Comment by tonynajjar** [comment]
@ooeygui FYI I am still adding changes to this PR as I am testing

**Comment by tonynajjar** [comment]
I'm done here :+1:

**Comment by tonynajjar** [comment]
Any estimate on when this can be merged? Are we talking days or weeks?

**Comment by ooeygui** [comment]
Running CI; although it seems to be targeting Foxy.

**Comment by ooeygui** [comment]
Pulled to bootstrap the humble builds. Thank you for your contribution!

**Pull request merged**